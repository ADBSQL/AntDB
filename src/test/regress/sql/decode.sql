set grammar to oracle;
set datestyle='ISO,YMD';
select decode(to_char(45),'3','9','45','3') from dual;
select decode(1.5*3,3.0,4,4.5,4.5) from dual;
select decode(3,4,4,3,6) from dual;
select decode('a','a','b','a') from dual;
select decode('你好','你好','b','a') from dual;
select decode(to_number('2'),2,3,'2',4) from dual;
select decode(to_number('2.4'),2.3,4,'2.4',4) from dual;
select decode(4,to_char('a'),4,'b',5) from dual;
select decode(to_char('5'),to_char('5'),to_char('6'),to_char('6'),to_char('7')) from dual;
select decode(3,exp(2),4,5) from dual;
select decode(exp(2),7,4,5) from dual;
select decode('3',tanh(3),4,5) from dual;
select decode('3',3,4,5) from dual;
select decode(3,'3',4,5) from dual;
select decode('3',3.0,4,5) from dual;
select decode('a','a','2',1,3) from dual;
select decode(to_date('2007-1-1','yyyy-mm-dd'),to_date('2007-1-1','yyyy-mm-dd'),1,0) from dual;
select decode(to_number('NaN'),'NaN',4,5) from dual;
select decode(exp(2),tanh(2),'4',exp(2),3) from dual;
select decode(to_date('2007-1-1','yyyy-mm-dd'),'2007-1-1',1,0) from dual;
select decode(to_date('2007-1-1','yyyy-mm-dd'),to_date('2007-1-1','yyyy-mm-dd'),1,0) from dual;
select decode(3,null,4,5) from dual;
select decode(3,'',4,5) from dual;
select decode(null,null,4,5) from dual;
select decode(null,3,4,5) from dual;
select decode(3,0,1,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,33,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,33,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,33,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,33,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,33,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3) from dual;
