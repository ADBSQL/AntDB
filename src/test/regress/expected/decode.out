set grammar to oracle;
set datestyle='ISO,YMD';
select decode(to_char(45),'3','9','45','3') from dual;
 DECODE(TO_CHAR(45),'3','9','45','3') 
--------------------------------------
 3
(1 row)

select decode(1.5*3,3.0,4,4.5,4.5) from dual;
 DECODE(1.5*3,3.0,4,4.5,4.5) 
-----------------------------
                         4.5
(1 row)

select decode(3,4,4,3,6) from dual;
 DECODE(3,4,4,3,6) 
-------------------
                 6
(1 row)

select decode('a','a','b','a') from dual;
 DECODE('A','A','B','A') 
-------------------------
 b
(1 row)

select decode('你好','你好','b','a') from dual;
 DECODE('你好','你好','B','A') 
-------------------------------
 b
(1 row)

select decode(to_number('2'),2,3,'2',4) from dual;
 DECODE(TO_NUMBER('2'),2,3,'2',4) 
----------------------------------
                                3
(1 row)

select decode(to_number('2.4'),2.3,4,'2.4',4) from dual;
 DECODE(TO_NUMBER('2.4'),2.3,4,'2.4',4) 
----------------------------------------
                                      4
(1 row)

select decode(4,to_char('a'),4,'b',5) from dual;
 DECODE(4,TO_CHAR('A'),4,'B',5) 
--------------------------------
                               
(1 row)

select decode(to_char('5'),to_char('5'),to_char('6'),to_char('6'),to_char('7')) from dual;
 DECODE(TO_CHAR('5'),TO_CHAR('5'),TO_CHAR('6'),TO_CHAR('6'),TO_CHAR('7')) 
--------------------------------------------------------------------------
 6
(1 row)

select decode(3,exp(2),4,5) from dual;
 DECODE(3,EXP(2),4,5) 
----------------------
                    5
(1 row)

select decode(exp(2),7,4,5) from dual;
 DECODE(EXP(2),7,4,5) 
----------------------
                    5
(1 row)

select decode('3',tanh(3),4,5) from dual;
 DECODE('3',TANH(3),4,5) 
-------------------------
                       5
(1 row)

select decode('3',3,4,5) from dual;
 DECODE('3',3,4,5) 
-------------------
                 4
(1 row)

select decode(3,'3',4,5) from dual;
 DECODE(3,'3',4,5) 
-------------------
                 4
(1 row)

select decode('3',3.0,4,5) from dual;
 DECODE('3',3.0,4,5) 
---------------------
                   4
(1 row)

select decode('a','a','2',1,3) from dual;
 DECODE('A','A','2',1,3) 
-------------------------
                       2
(1 row)

select decode(to_date('2007-1-1','yyyy-mm-dd'),to_date('2007-1-1','yyyy-mm-dd'),1,0) from dual;
 DECODE(TO_DATE('2007-1-1','YYYY-MM-DD'),TO_DATE('2007-1-1','YYYY-MM-DD'),1,0) 
-------------------------------------------------------------------------------
                                                                             1
(1 row)

select decode(to_number('NaN'),'NaN',4,5) from dual;
 DECODE(TO_NUMBER('NAN'),'NAN',4,5) 
------------------------------------
                                  4
(1 row)

select decode(exp(2),tanh(2),'4',exp(2),3) from dual;
 DECODE(EXP(2),TANH(2),'4',EXP(2),3) 
-------------------------------------
                                   3
(1 row)

select decode(to_date('2007-1-1','yyyy-mm-dd'),'2007-1-1',1,0) from dual;
 DECODE(TO_DATE('2007-1-1','YYYY-MM-DD'),'2007-1-1',1,0) 
---------------------------------------------------------
                                                       0
(1 row)

select decode(to_date('2007-1-1','yyyy-mm-dd'),to_date('2007-1-1','yyyy-mm-dd'),1,0) from dual;
 DECODE(TO_DATE('2007-1-1','YYYY-MM-DD'),TO_DATE('2007-1-1','YYYY-MM-DD'),1,0) 
-------------------------------------------------------------------------------
                                                                             1
(1 row)

select decode(3,null,4,5) from dual;
 DECODE(3,NULL,4,5) 
--------------------
                  5
(1 row)

select decode(3,'',4,5) from dual;
 DECODE(3,'',4,5) 
------------------
                5
(1 row)

select decode(null,null,4,5) from dual;
 DECODE(NULL,NULL,4,5) 
-----------------------
                     4
(1 row)

select decode(null,3,4,5) from dual;
 DECODE(NULL,3,4,5) 
--------------------
                  5
(1 row)

select decode(3,0,1,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,33,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,33,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,33,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,33,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,33,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3) from dual;
 DECODE(3,0,1,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,33,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,33,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,33,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,33,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,33,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3) 
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        3
(1 row)

